feat(RELEASE-1001): update MintMaker utils image rules

seanconroy2021 opened the pull request:
## Describe your changes
Update Renovate config to manage the following images:
* quay.io/konflux-ci/release-service-utils
* quay.io/conforma/cli
* quay.io/redhat-appstudio/build-trusted-artifacts
* quay.io/konflux-ci/appstudio-utils
* quay.io/konflux-ci/charon
* amazon/aws-cli

Paths of images to update:
* tasks/managed
* tasks/internal
* tasks/collectors
* stepActions
* integration-tests/pipelines

StepActions are currently unsupported by the tekton manager, so using the kubernetes manager as a workaround.

Each image gets its own PR, with MintMaker creating PRs between 05:00 and 09:00 UTC each day.

Testing: updating all utils images took 2h27m to complete unit tests.

## Relevant Jira
[RELEASE-1001](https://issues.redhat.com/browse/RELEASE-1001)
## Checklist before requesting a review
- [ ] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [ ] My commit message includes `Signed-off-by: My name <email>`
- [ ] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [ ] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

seanconroy2021 commented:
MintMaker team will be adding support for StepActions at some point reached out to them https://issues.redhat.com/browse/CWFHEALTH-4575

swickersh commented on renovate.json:
```diff
@@ -1,6 +1,46 @@
+  "rebaseWhen": "behind-base-branch",
   "tekton": {
-    "schedule": ["* 0 * * *"]
+    "schedule": ["* 0 * * *"],
```
Can you please add a small comment above the schedule to remind us the difference between the global time schedule and at the package level below? And maybe brief reasoning for the times selected.

seanconroy2021 replied:
I tried adding inline comments to the Renovate config, but since it is being validated, using # or // <text> causes JSON linting when running `npx --yes --package renovate -- renovate-config-validator`.

I could add it to the description for each if you think it will be clearer.

swickersh replied:
ah no worries then. All good

swickersh commented on renovate.json:
```diff
@@ -1,6 +1,46 @@
+        "matchFileNames": [
+          "tasks/managed/**",
+          "tasks/internal/**",
+          "tasks/collectors/**"
```
Why not just tasks? That's the only three directories:
```suggestion
          "tasks/**"
```

seanconroy2021 replied:
fixed ty :)

swickersh reviewed (commented):
After about 40 files, GitHub is making me click "Load Diff" for every one because this is such a large change. Are the rest of them all just updating the tag to the digest?

seanconroy2021 commented:
/retest

seanconroy2021 committed:
feat(RELEASE-1001): update MintMaker utils image rules

Update Renovate config to manage the following images:
* quay.io/konflux-ci/release-service-utils
* quay.io/conforma/cli
* quay.io/redhat-appstudio/build-trusted-artifacts
* quay.io/konflux-ci/appstudio-utils
* quay.io/konflux-ci/charon
* amazon/aws-cli

Paths of images to update:
* tasks/managed
* tasks/internal
* tasks/collectors
* stepActions
* integration-tests/pipelines

StepActions are currently unsupported by the tekton manager,
so using the kubernetes manager as a workaround.

Each image gets its own PR, with MintMaker creating PRs
between 05:00 and 09:00 UTC each day.

Testing: updating all utils images took 2h27m to complete unit tests.

Assisted-by: Cursor
Signed-off-by: Sean Conroy <sconroy@redhat.com>
